pkg/api/controller.go

```diff
@@ -1714,7 +1736,12 @@ func (c *Controller) UploadObject(w http.ResponseWriter, r *http.Request, reposi
 		Size:            blob.Size,
 		Checksum:        blob.Checksum,
 	}
-	err = c.Catalog.CreateEntry(ctx, repo.Name, branch, entry)
+
+	err = c.Catalog.CreateEntry(ctx, repo.Name, branch, entry, graveler.IfAbsent(!allowOverwrite))
```

Comment:
Cool, double-checked locking actually works here!  (The previous test is only an optimization, but still nice to know.)